// @analogjs/vite-plugin-angular/src/lib/tools/src/builders/vite/vite-build.impl.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const architect_1 = require("@angular-devkit/architect");
async function viteBuilder(options, context) {
    const { createBuilder } = await Function('return import("vite")')();
    const projectConfig = await context.getProjectMetadata(context.target);
    const projectName = context.target.project;
    const configuration = context.target?.configuration || 'production';
    const buildTargetSpecifier = `::${configuration}`;
    const buildTarget = (0, architect_1.targetFromTargetString)(buildTargetSpecifier, projectName, 'build');
    const browserBuilderName = await context.getBuilderNameForTarget(buildTarget);
    const rawBuildOptions = await context.getTargetOptions(buildTarget);
    const buildOptions = await context.validateOptions(rawBuildOptions, browserBuilderName);
    const buildConfig = {
        configFile: options.configFile,
        root: projectConfig.root,
        mode: (process.env.NODE_ENV ??
            buildOptions.mode ??
            configuration),
        build: {
            outDir: options.outputPath,
            sourcemap: !!buildOptions.sourcemap,
        },
    };
    try {
        const builder = await createBuilder(buildConfig, false);
        await builder.buildApp();
        return {
            success: true,
        };
    }
    catch (e) {
        console.error(e);
        return {
            success: false,
        };
    }
}
exports.default = (0, architect_1.createBuilder)(viteBuilder);
