// @analogjs/vite-plugin-angular/src/lib/tools/src/builders/vite-dev-server/dev-server.impl.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const architect_1 = require("@angular-devkit/architect");
async function viteDevServerBuilder(options, context) {
    const { createServer } = await Function('return import("vite")')();
    const projectConfig = await context.getProjectMetadata(context.target);
    const projectName = context.target.project;
    const buildTargetSpecifier = options.buildTarget ?? `::development`;
    const buildTarget = (0, architect_1.targetFromTargetString)(buildTargetSpecifier, projectName, 'build');
    const browserBuilderName = await context.getBuilderNameForTarget(buildTarget);
    const rawBuildOptions = await context.getTargetOptions(buildTarget);
    const buildOptions = await context.validateOptions(rawBuildOptions, browserBuilderName);
    const serverConfig = {
        configFile: buildOptions.configFile,
        root: projectConfig.root,
        mode: (process.env.NODE_ENV ??
            buildOptions.mode ??
            'development'),
        build: {
            sourcemap: !!buildOptions.sourcemap,
        },
        server: {
            hmr: options?.hmr,
            port: options?.port,
        },
    };
    try {
        const server = await createServer(serverConfig);
        await runViteDevServer(server);
        const resolvedUrls = [
            ...server.resolvedUrls.local,
            ...server.resolvedUrls.network,
        ];
        await new Promise((resolve) => {
            process.once('SIGINT', () => resolve());
            process.once('SIGTERM', () => resolve());
            process.once('exit', () => resolve());
        });
        return {
            success: true,
            baseUrl: resolvedUrls[0] ?? '',
        };
    }
    catch (e) {
        console.error(e);
        return {
            success: false,
            baseUrl: '',
        };
    }
}
// vite ViteDevServer
async function runViteDevServer(server) {
    await server.listen();
    server.printUrls();
    const processOnExit = async () => {
        await server.close();
    };
    process.once('SIGINT', processOnExit);
    process.once('SIGTERM', processOnExit);
    process.once('exit', processOnExit);
}
exports.default = (0, architect_1.createBuilder)(viteDevServerBuilder);
